launch_ros/launch_ros/actions/load_composable_nodes.py

```diff
@@ -161,10 +159,17 @@ def execute(
         context: LaunchContext
     ) -> Optional[List[Action]]:
         """Execute the action."""
+        # resolve target container node name
+        if isinstance(self.__target_container, ComposableNodeContainer):
+            self.__final_target_container_name = self.__target_container.node_name
+        else:  # TODO: maybe should do better type checking here
+            subs = normalize_to_list_of_substitutions(self.__target_container)
+            self.__final_target_container_name = perform_substitutions(context, subs)
```

Comment:
@eric1221bday why not resolving this TODO now?

Reply:
@hidmic So I meant to revisit this, the core of the issue is the same reason I removed the ensure_argument_type at the top, which is that Unions are tricky to type check. There's a stackoverflow [here](https://stackoverflow.com/a/49471187) with a suggested way of checking using both the `__origin__` and `__args__` attribute, but I'm unsure how advisable that is.

Reply:
As I said above you can use the types tuple we have, but if you choose to leave the TODO, please put your github handle into it (that's our convention): `# TODO(eric1221bday): ...`

Reply:
@wjwwood alright I ended up using `is_a_subclass` for the ComposableNodeContainer and `isinstance` with `SomeSubstitutionsType_types_tuple` to do the type checking. If the argument is neither I will log and error and return. Considering the type checking at the constructor this is a bit of an unlikely hit, but no harm really.